qa/tasks/cephfs/test_nfs.py

```diff
@@ -613,6 +613,56 @@ def disable_libcephfs_logging(self, cluster_name, ganesha_spec, conf_path, ganes
             # restart ganesha daemon for the changes to take effect
             self._orch_cmd("restart", f"nfs.{cluster_name}")
 
+    def set_subvolume_snapshot_visibility_client_config(self, flag):
+        client_visibility_conf = 'client_respect_subvolume_snapshot_visibility'
+        self._cmd('config', 'set', 'client', client_visibility_conf, flag)
+
+    def get_subvolume_snapshot_visibility_client_config(self, flag):
+        client_visibility_conf = 'client_respect_subvolume_snapshot_visibility'
+        client_respects_visibility = self._cmd('config', 'get', 'client',
+                                               client_visibility_conf).strip()
+        self.assertEqual(client_respects_visibility, flag)
+
+    def set_subvolume_snapshot_visibility(self, volume, subvolume, flag):
+        self._cmd('fs', 'subvolume', 'snapshot_visibility', 'set', volume,
+                  subvolume, flag)
+
+    def ensure_subvolume_snapshot_visibility(self, volume, subvolume, flag):
+        snapshots_visible = self._cmd('fs', 'subvolume', 'snapshot_visibility',
+                                      'get', volume, subvolume).strip()
+        self.assertEqual(snapshots_visible, flag)
+
+    def lookup_snapshot(self, snapshot_ls_cmd, snapshot, negative_lookup=False):
+        with contextutil.safe_while(sleep=1, tries=15) as proceed:
+            while proceed():
+                try:
+                    snapshot_ls = self.ctx.cluster.run(args=snapshot_ls_cmd,
+                                                       stdout=StringIO(),
+                                                       stderr=StringIO())
+                    snapshot_ls = snapshot_ls[0].stdout.getvalue().strip()
+                    # not an expected behaviour, something's wrong with cluster
+                    if len(snapshot_ls) < 1:
+                        raise Exception(f'{snapshot_ls_cmd} returned empty')
+                    if negative_lookup:
+                        self.assertNotIn(snapshot, snapshot_ls)
+                    else:
+                        self.assertIn(snapshot, snapshot_ls)
+                    break
+                except AssertionError:
+                    if negative_lookup:
+                        log.info(f'snapshot {snapshot} still visible, '
```

Comment:
If retrying, does this needs to be at debug log level?

(and the message in the else case too)

Reply:
sure